src/librustc/hir/map/mod.rs

```diff
@@ -613,6 +673,21 @@ impl<'hir> Map<'hir> {
         result
     }
 
+    // FIXME(@ljedrz): replace the NodeId variant
+    pub fn find_by_hir_id(&self, hir_id: HirId) -> Option<Node<'hir>> {
+        let result = self.find_entry_by_hir_id(hir_id).and_then(|entry| {
+            if let Node::Crate = entry.node {
+                None
+            } else {
+                Some(entry.node)
+            }
+        });
+        if result.is_some() {
+            self.read_by_hir_id(hir_id);
+        }
+        result
+    }
+
```

Comment:
This should just convert the `HirId` to a `NodeId` and call the regular `find` method.